### PR TITLE
Docker Build

### PR DESCRIPTION
**Pull request by dtan4**
## WHY

To build Docker image in risu
## WHAT
- Add cache to cloned repository
- Run `docker build`

### CODE REVIEW

**Comment by dtan4** [comment]
Building...

``` bash
core@core-01 ~ $ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
<none>              <none>              634fdcc9a086        34 seconds ago      717.1 MB
ruby                2.2.2               9664620d4c2a        39 hours ago        717.1 MB
risu_web            latest              721ad8f2f04e        3 days ago          6.251 MB
<none>              <none>              ca74ea00d80e        3 days ago          14.52 MB
<none>              <none>              e1bf426ced24        3 days ago          14.52 MB
<none>              <none>              1d748a88ddd9        3 days ago          5.951 MB
<none>              <none>              b31c97fd2765        3 days ago          5.31 MB
gliderlabs/alpine   3.2                 5bd56d818842        8 weeks ago         5.25 MB
```

**Comment by koudaiii** [comment]
risu.go の　dockerPushも合わせて移動したい。。。

**Comment by dtan4** [comment]
@wantedly/infrastructure plz review this :bow:

File: cache/cache.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ type Cache interface {
 }
 
 const (
+	CacheBaseDir            = "/var/risu/cache/"
 	DefaultInflatedCacheDir = "/tmp/risu/cache"
```
Comment (awakia) [inline cache/cache.go]: How about using with tailing slash like `"/tmp/risu/cache/"`?

File: docker.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				return err
+			}
+		}
+	}
+
+	var dockerEndpoint string
+
+	if os.Getenv("DOCKER_HOST") != "" {
+		dockerEndpoint = os.Getenv("DOCKER_HOST")
+	}
```
Comment (awakia) [inline docker.go]: Why don't you write previous 4 lines in one sentence?

```
dockerEndpoint := os.Getenv("DOCKER_HOST")
```

File: docker.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/wantedly/risu/schema"
+)
+
+const (
+	DefaultDockerEndpoint = "unix:///var/run/docker.sock"
+)
+
+func dockerBuild(build schema.Build) error {
+	clonePath := DefaultSourceBaseDir + build.SourceRepo
+	cache := c.NewCache(os.Getenv("CACHE_BACKEND"))
```
Comment (spesnova) [inline docker.go]: How about extract these 22~44 lines to `addCacheToSrcRepo()`.
I think it can be more clear that what this lines is doing.

**Comment by spesnova** [comment]
LGTM :+1:

**Pull request merged**